index.js

```diff
@@ -56,34 +57,87 @@ function HyperString (db, opts) {
   })
 }
 
-HyperString.prototype.insert = function (prev, chr, cb) {
-  // TODO: check that it's a single character
-  // TODO: support inserting a whole string, and breaking it up into
-  // one-char inserts
-  var op = {
-    op: 'insert',
-    chr: chr,
-    prev: prev || null
+HyperString.prototype.insert = function (prev, string, done) {
```

Comment:
*New Exciting Failure Mode*: errors that cause partial inserts and deletes. An atomic batch operation is ideal -- I'm not sure if [`log.batch`](https://github.com/mafintosh/hyperlog#logbatchdocs-opts-cb) can help us, since there's a dependency chain here.

I wonder if @mafintosh :wave: knows how we can achieve this.